// @share-package/common/build/events/types/permissions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPermission = void 0;
var ListPermission;
(function (ListPermission) {
    // Create, read, update, delete orders
    ListPermission["OrderCreate"] = "order.create";
    ListPermission["OrdersRead"] = "order.read";
    ListPermission["OrderUpdate"] = "order.update";
    ListPermission["OrderDelete"] = "order.delete";
    // Create, read, update, delete shifts
    ListPermission["ShiftCreate"] = "shift.create";
    ListPermission["ShiftRead"] = "shift.read";
    ListPermission["ShiftUpdate"] = "shift.update";
    ListPermission["ShiftDelete"] = "shift.delete";
    // Create, read, update, delete employeee
    ListPermission["EmployeeCreate"] = "employee.create";
    ListPermission["EmployeeRead"] = "employee.read";
    ListPermission["EmployeeUpdate"] = "employee.update";
    ListPermission["EmployeeDelete"] = "employee.delete";
    // Create, read, update, delete customers
    ListPermission["CustomerCreate"] = "customer.create";
    ListPermission["CustomerRead"] = "customer.read";
    ListPermission["CustomerUpdate"] = "customer.update";
    ListPermission["CustomerDelete"] = "customer.delete";
    // Create, read, update, delete roles
    ListPermission["RoleCreate"] = "role.create";
    ListPermission["RoleRead"] = "role.read";
    ListPermission["RoleUpdate"] = "role.update";
    ListPermission["RoleDelete"] = "role.delete";
    //
    // Create, read, update, delete roles
    ListPermission["RolePermissionCreate"] = "role-permssion.create";
    ListPermission["RolePermissionRead"] = "role-permssion.read";
    ListPermission["RolePermissionDelete"] = "role-permssion.delete";
    // Create, read, update, delete services
    ListPermission["ServiceCreate"] = "service.create";
    ListPermission["ServiceRead"] = "service.read";
    ListPermission["ServiceUpdate"] = "service.update";
    ListPermission["ServiceDelete"] = "service.delete";
    // Create, read, update, delete package service
    ListPermission["PackageCreate"] = "package.create";
    ListPermission["PackageRead"] = "package.read";
    ListPermission["PackageUpdate"] = "package.update";
    ListPermission["PackageDelete"] = "package.delete";
    // Create, read, update, delete package product
    ListPermission["ProductCreate"] = "product.create";
    ListPermission["ProductRead"] = "product.read";
    ListPermission["ProductUpdate"] = "product.update";
    ListPermission["ProductDelete"] = "product.delete";
    //
    ListPermission["UserShiftCreate"] = "user-shift.create";
    ListPermission["UserShiftRead"] = "user-shift.read";
    ListPermission["UserShiftUpdate"] = "user-shift.update";
    ListPermission["UserShiftDelete"] = "user-shift.delete";
    ListPermission["BranchCreate"] = "branch.create";
    ListPermission["BranchRead"] = "branch.read";
    ListPermission["BranchUpdate"] = "branch.update";
    ListPermission["BranchDelete"] = "branch.delete";
})(ListPermission || (exports.ListPermission = ListPermission = {}));
